cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlApp.java

```diff
@@ -0,0 +1,176 @@
+    _jmxReporter.start();
+
+    _kafkaCruiseControl = new AsyncKafkaCruiseControl(config, metricRegistry);
+
+    _server = new Server();
+    NCSARequestLog requestLog = createRequestLog();
+    if (requestLog != null) {
+      _server.setRequestLog(requestLog);
+    }
+    _server.setConnectors(new Connector[]{ setupHttpConnector(hostname, port) });
+
+    ServletContextHandler contextHandler = createContextHandler();
+    ConstraintSecurityHandler securityHandler = createSecurityHandler();
+    if (securityHandler != null) {
+      contextHandler.setSecurityHandler(securityHandler);
+    }
+    _server.setHandler(contextHandler);
+
+    setupWebUi(contextHandler);
+
+    KafkaCruiseControlServlet servlet = new KafkaCruiseControlServlet(_kafkaCruiseControl, metricRegistry);
+    String apiUrlPrefix = config.getString(KafkaCruiseControlConfig.WEBSERVER_API_URLPREFIX);
+    ServletHolder servletHolder = new ServletHolder(servlet);
+    contextHandler.addServlet(servletHolder, apiUrlPrefix);
+  }
+
+  void start() throws Exception {
+    _kafkaCruiseControl.startUp();
+    _server.start();
+    printStartupInfo();
+  }
+
+  void registerShutdownHook() {
+    Runtime.getRuntime().addShutdownHook(new Thread(this::stop));
+  }
+
+  void stop() {
+    _kafkaCruiseControl.shutdown();
+    _jmxReporter.close();
+  }
+
+  public String serverUrl() {
+    return _server.getURI().toString();
+  }
+
+  private void printStartupInfo() {
+    boolean corsEnabled = _config.getBoolean(KafkaCruiseControlConfig.WEBSERVER_HTTP_CORS_ENABLED_CONFIG);
+    String webApiUrlPrefix = _config.getString(KafkaCruiseControlConfig.WEBSERVER_API_URLPREFIX);
+    String uiUrlPrefix = _config.getString(KafkaCruiseControlConfig.WEBSERVER_UI_URLPREFIX);
+    String webDir = _config.getString(KafkaCruiseControlConfig.WEBSERVER_UI_DISKPATH);
+    String sessionPath = _config.getString(KafkaCruiseControlConfig.WEBSERVER_SESSION_PATH);
+    System.out.println(">> ********************************************* <<");
+    System.out.println(">> Application directory            : " + System.getProperty("user.dir"));
+    System.out.println(">> REST API available on            : " + webApiUrlPrefix);
+    System.out.println(">> Web UI available on              : " + uiUrlPrefix);
+    System.out.println(">> Web UI Directory                 : " + webDir);
+    System.out.println(">> Cookie prefix path               : " + sessionPath);
+    System.out.println(">> Kafka Cruise Control started on  : " + serverUrl());
+    System.out.println(">> CORS Enabled ?                   : " + corsEnabled);
+    System.out.println(">> ********************************************* <<");
+  }
+
+  private ServerConnector setupHttpConnector(String hostname, int port) {
+    String keyStoreLocation = _config.getString(KafkaCruiseControlConfig.WEBSERVER_SSL_KEYSTORE_LOCATION);
+    ServerConnector serverConnector;
+    if (keyStoreLocation != null) {
```

Comment:
I think it'd make sense to enable SSL / authentication independently of each other, I just thought I wouldn't introduce an extra config for `WEBSERVER_SSL_ENABLE` but it would be symmetrical to the authentication bit, so afterall it'd make sense. I'll change the code according to this and also add the sanity check.